PostgreSQL formulaes: Fix displayed data dir in caveats

cypher opened the pull request:
The versioned PostgreSQL formulaes use a versioned data dir, so they can coexist on the same system. However, the data dir path displayed in `caveats` was not correct.

Also, this updated the `plist_options` call to use the name of the formula instead of hardcoding it.

Note: PostgreSQL@12 is not included in this PR, as this is fixed in #65102


-----
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

gromgit commented:
While you're fixing up the caveats in the versioned formulae, I suggest also removing the entire section that mentions `brew postgresql-upgrade-database`. The steps it takes are only necessary when going from one major PostgreSQL version to another, but each versioned formula is already limited to its own major version, so running a DB upgrade is a time-wasting no-op at best.

Only the main `postgresql` formula needs to mention this tool.

cypher commented:
@gromgit I've removed all the upgrade notices. The ones for 9.x don't mention `brew postgresql-upgrade-database`, but I've removed them anyway for the same reason.

cypher commented:
Also, I've not updated the PostgreSQL@12 caveat, since this should probably also be part of #65102.

cypher commented:
Closing, as this PR is being folded into https://github.com/Homebrew/homebrew-core/pull/65102.
